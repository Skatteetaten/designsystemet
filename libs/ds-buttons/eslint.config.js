import nxPlugin from '@nx/eslint-plugin';

import baseConfig from '../../eslint.config.js';

export default [
  ...baseConfig,
  ...nxPlugin.configs['flat/react'],
  {
    ignores: ['!**/*', '**/storybook-static'],
    files: ['**/*.ts', '**/*.tsx', '**/*.js', '**/*.jsx'],
    rules: {},
  },
  {
    ignores: ['!**/*', '**/storybook-static'],
    files: ['**/*.ts', '**/*.tsx'],
    rules: {},
  },
  {
    ignores: ['!**/*', '**/storybook-static'],
    files: ['**/*.js', '**/*.jsx'],
    rules: {},
  },
];
