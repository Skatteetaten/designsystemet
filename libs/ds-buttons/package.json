{
  "name": "@skatteetaten/ds-buttons",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.5.1",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "type": "module",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "@skatteetaten/ds-core-utils": "1.5.1",
    "@skatteetaten/ds-icons": "1.5.1",
    "@skatteetaten/ds-progress": "1.5.1",
    "react": "18.3.1",
    "react-i18next": "14.1.3",
    "i18next": "23.16.8"
  },
  "scripts": {},
  "dependencies": {},
  "devDependencies": {}
}
