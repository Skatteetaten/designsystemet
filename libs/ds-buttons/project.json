{
  "name": "ds-buttons",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-buttons/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-buttons",
        "tsConfig": "libs/ds-buttons/tsconfig.lib.json",
        "project": "libs/ds-buttons/package.json",
        "entryFile": "libs/ds-buttons/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-buttons/rollup.config.js"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-buttons/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ds-buttons"],
      "options": {
        "jestConfig": "libs/ds-buttons/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-buttons/**/*.css",
          "libs/ds-buttons/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      }
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-buttons"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-buttons {args.switches}"
      }
    }
  }
}
