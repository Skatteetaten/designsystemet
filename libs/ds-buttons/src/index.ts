export * from './Button/Button';
export * from './Button/Button.types';
export * from './IconButton/IconButton';
export * from './IconButton/IconButton.types';
export * from './InlineButton/InlineButton';
export * from './InlineButton/InlineButton.types';
export * from './Link/Link';
export * from './Link/Link.types';
export * from './LinkGroup/LinkGroup';
export * from './LinkGroup/LinkGroup.types';
export * from './MegaButton/MegaButton';
export * from './MegaButton/MegaButton.types';
export * from './ScrollToTopButton/ScrollToTopButton';
export * from './ScrollToTopButton/ScrollToTopButton.types';
