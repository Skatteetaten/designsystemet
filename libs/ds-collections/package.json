{
  "name": "@skatteetaten/ds-collections",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.4.0",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14",
    "@skatteetaten/ds-core-utils": "1.3.0",
    "@skatteetaten/ds-icons": "1.3.0",
    "@skatteetaten/ds-buttons": "1.3.0",
    "@skatteetaten/ds-content": "1.3.0",
    "@skatteetaten/ds-typography": "1.3.0",
    "@skatteetaten/ds-progress": "1.3.0"
  },
  "scripts": {}
}
