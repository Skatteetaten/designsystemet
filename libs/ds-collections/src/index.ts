export * from './Accordion/Accordion';
export * from './Accordion/Accordion.types';
export * from './AccordionItem/AccordionItem.types';
export * from './Chips/Chips';
export * from './Chips/Chips.types';
export * from './ChipsToggle/ChipsToggle';
export * from './ChipsToggle/ChipsToggle.types';
export * from './ChipsRemovable/ChipsRemovable';
export * from './ChipsRemovable/ChipsRemovable.types';
export * from './OpenClose/OpenClose';
export * from './OpenClose/OpenClose.types';
export * from './StepList/StepList';
export * from './StepList/StepList.types';
export * from './StepListStep/StepListStep.types';
export * from './Tabs/Tabs';
export * from './Tabs/Tabs.types';
