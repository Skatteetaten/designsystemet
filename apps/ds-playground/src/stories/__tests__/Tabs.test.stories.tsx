import { JSX, useState } from 'react';

import { useArgs } from '@storybook/preview-api';
import { Meta, StoryFn, StoryObj } from '@storybook/react';
import { expect, fn, userEvent, waitFor, within } from '@storybook/test';

import { Button } from '@skatteetaten/ds-buttons';
import { Tabs, TabsProps } from '@skatteetaten/ds-collections';

const meta = {
  component: Tabs,
  title: 'Tester/Tabs/Tabs',
  argTypes: {
    // Baseprops
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    defaultValue: { table: { disable: true } },
    value: { table: { disable: true } },
    isMultiline: { table: { disable: true } },
    variant: { table: { disable: true } },
    hasBorder: { table: { disable: true } },
    // Events
    onChange: { table: { disable: true } },
  },
} satisfies Meta<typeof Tabs>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultArgs: TabsProps = {
  onChange: fn(),
  defaultValue: 'tab1',
};

const TemplateTabs: StoryFn<typeof Tabs> = (args) => {
  return (
    <Tabs
      {...args}
      onChange={(e): void => {
        args.onChange?.(e);
      }}
    >
      <Tabs.List>
        <Tabs.Tab value={'tab1'}>{'Person'}</Tabs.Tab>
        <Tabs.Tab value={'tab2'}>{'Bedrift'}</Tabs.Tab>
        <Tabs.Tab value={'tab3'}>{'Organisasjon'}</Tabs.Tab>
      </Tabs.List>
      <Tabs.Panel value={'tab1'}>{'Tabs.Panel Person'}</Tabs.Panel>
      <Tabs.Panel value={'tab2'}>{'Tabs.Panel Bedrift'}</Tabs.Panel>
      <Tabs.Panel value={'tab3'}>{'Tabs.Panel Organisasjon'}</Tabs.Panel>
    </Tabs>
  );
};

const TemplateTabsWithOnClick: StoryFn<typeof Tabs> = (args) => {
  const [value, setValue] = useState('tab1');

  const handleOnClick = (v: string): void => {
    setValue(v);
  };

  return (
    <Tabs {...args} value={value}>
      <Tabs.List>
        <Tabs.Tab value={'tab1'} onClick={() => handleOnClick('tab1')}>
          {'Person'}
        </Tabs.Tab>
        <Tabs.Tab value={'tab2'} onClick={() => handleOnClick('tab2')}>
          {'Bedrift'}
        </Tabs.Tab>
        <Tabs.Tab value={'tab3'} onClick={() => handleOnClick('tab3')}>
          {'Organisasjon'}
        </Tabs.Tab>
      </Tabs.List>
      <Tabs.Panel value={'tab1'}>{'Tabs.Panel Person'}</Tabs.Panel>
      <Tabs.Panel value={'tab2'}>{'Tabs.Panel Bedrift'}</Tabs.Panel>
      <Tabs.Panel value={'tab3'}>{'Tabs.Panel Organisasjon'}</Tabs.Panel>
    </Tabs>
  );
};

export const Defaults = {
  name: 'Defaults (A2)',
  render: TemplateTabs,
  args: {
    ...defaultArgs,
  },
  argTypes: {
    defaultValue: { table: { disable: false } },
  },
  parameters: {
    viewport: {
      defaultViewport: undefined,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const tab = canvas.getByRole('tab', { name: 'Person' });
    await expect(tab).toBeInTheDocument();
  },
} satisfies Story;

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLDivElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: { imageSnapshot: { disable: true } },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const tabs = canvas.getAllByRole('generic')[1];
    await expect(tabs).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5, B1)',
  render: TemplateTabs,
  args: {
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
    defaultValue: 'tab1',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toHaveAttribute('id', 'htmlId');
    await expect(container).toHaveClass('dummyClassname');
    await expect(container).toHaveAttribute('lang', 'nb');
    await expect(container).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const WithVariantCompact = {
  name: 'With Variant Compact (A1)',
  render: TemplateTabs,
  args: {
    ...defaultArgs,
    variant: 'compact',
  },
  argTypes: {
    variant: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: false },
  },
} satisfies Story;

export const WithDefaultValue = {
  name: 'With DefaultValue',
  render: TemplateTabs,
  args: {
    ...defaultArgs,
    defaultValue: 'tab2',
    onChange: fn(),
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const secondTab = canvas.getByRole('tab', {
      name: 'Bedrift',
    });
    await expect(secondTab).toBeInTheDocument();
    await expect(secondTab).toHaveAttribute('aria-selected', 'true');
  },
} satisfies Story;

export const WithValue = {
  name: 'With Value',
  args: {
    ...defaultArgs,
    defaultValue: undefined,
    value: 'tab2',
  },
  argTypes: {
    value: { table: { disable: false } },
  },
  render: (args): JSX.Element => {
    const [{ value }, updateArgs] = useArgs();
    const toggleTab = (): void => {
      if (value === 'tab1') {
        updateArgs({ value: 'tab2' });
      } else {
        updateArgs({ value: 'tab1' });
      }
    };
    return (
      <>
        <Tabs {...args} value={value}>
          <Tabs.List>
            <Tabs.Tab value={'tab1'}>{'Person'}</Tabs.Tab>
            <Tabs.Tab value={'tab2'}>{'Bedrift'}</Tabs.Tab>
          </Tabs.List>
          <Tabs.Panel value={'tab1'}>{'Tabs.Panel - Person'}</Tabs.Panel>
          <Tabs.Panel value={'tab2'}>{'Tabs.Panel - Bedrift'}</Tabs.Panel>
        </Tabs>
        <Button onClick={toggleTab}>{'ToggleTab'}</Button>
      </>
    );
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement, step }): Promise<void> => {
    const canvas = within(canvasElement);
    await step(
      'Sjekker om Bedrift-tab finnes og har attribut aria-selected:true',
      async () => {
        const secondTab = await canvas.findByRole('tab', {
          name: 'Bedrift',
          selected: true,
        });
        await expect(secondTab).toBeInTheDocument();
      }
    );

    await step(
      'Endrer value-prop utenfra og forventer at Person-tab er aktiv',
      async () => {
        const button = await canvas.findByRole('button', { name: 'ToggleTab' });
        await userEvent.click(button);
        const firsttab = await canvas.findByRole('tab', {
          name: 'Person',
          selected: true,
        });
        await expect(firsttab).toBeInTheDocument();
      }
    );

    await step(
      'Ingen test - Nullstiller - Toggler aktiv tab tilbake til tab2/Bedrift for å kunne kjøre test i nettelser flere ganger',
      async () => {
        const button = await canvas.findByRole('button', { name: 'ToggleTab' });
        await userEvent.click(button);
      }
    );
  },
} satisfies Story;

export const WithAriaRolesTabindex = {
  name: 'With Aria Roles tabIndex (B1, B2)',
  render: TemplateTabs,
  args: {
    ...defaultArgs,
    onChange: fn(),
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement, step }): Promise<void> => {
    const canvas = within(canvasElement);
    const firstTab = canvas.getByRole('tab', {
      name: 'Person',
    });
    await expect(firstTab).toBeInTheDocument();
    const tabList = canvas.getByRole('tablist');
    await expect(tabList).toBeInTheDocument();
    const tabPanel = canvas.getByRole('tabpanel');
    await expect(tabPanel).toBeInTheDocument();
    await step(
      'Sjekk om korrekt tabIndex på aktiv og ikke-aktive tabs',
      async () => {
        const tabs = canvas.getAllByRole('tab');
        Object.entries(tabs).forEach(async ([_i, tabElement], index) => {
          await expect(tabElement).toHaveAttribute(
            'tabIndex',
            index === 0 ? '0' : '-1'
          );
        });
      }
    );
  },
} satisfies Story;

export const WithMultiline = {
  name: 'With Multiline (A5)',
  args: {
    ...defaultArgs,
    isMultiline: true,
  },
  argTypes: {
    isMultiline: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: false },
    viewport: {
      defaultViewport: '--mobile',
    },
  },
  render: (args): JSX.Element => {
    return (
      <Tabs {...args}>
        <Tabs.List>
          <Tabs.Tab value={'tab1'}>{'Person'}</Tabs.Tab>
          <Tabs.Tab value={'tab2'}>{'Bedrift'}</Tabs.Tab>
          <Tabs.Tab value={'tab3'}>{'Ubehandlede saker (+99)'}</Tabs.Tab>
          <Tabs.Tab value={'tab4'}>{'Behandlede saker (1)'}</Tabs.Tab>
        </Tabs.List>
        <Tabs.Panel value={'tab1'}>{'Tabs.Panel - Person'}</Tabs.Panel>
        <Tabs.Panel value={'tab2'}>{'Tabs.Panel - Bedrift'}</Tabs.Panel>
        <Tabs.Panel value={'tab3'}>
          {'Tabs.Panel - Ubehandlede saker (+99)'}
        </Tabs.Panel>
        <Tabs.Panel value={'tab4'}>{'Tabs.Panel - Behandlede'}</Tabs.Panel>
      </Tabs>
    );
  },
} satisfies Story;

export const WithTabClick = {
  name: 'With Tab Click (A2, B2, B3, B4)',
  render: TemplateTabs,
  args: {
    ...defaultArgs,
    onChange: fn(),
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ args, canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const firstTab = canvas.getByRole('tab', {
      name: 'Person',
    });
    await expect(firstTab).toBeInTheDocument();
    await expect(firstTab).toHaveAttribute('aria-selected', 'true');
    await expect(firstTab).toHaveAttribute('tabIndex', '0');
    const secondTab = canvas.getByRole('tab', {
      name: 'Bedrift',
    });
    await userEvent.click(secondTab);
    await waitFor(() => expect(args.onChange).toHaveBeenCalled());
    await expect(firstTab).toHaveAttribute('aria-selected', 'false');
    await expect(firstTab).toHaveAttribute('tabIndex', '-1');
  },
} satisfies Story;

export const WithTabOnClickEvent = {
  name: 'With Tab OnClick Event',
  render: TemplateTabsWithOnClick,
  args: {
    ...defaultArgs,
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const firstTab = canvas.getByRole('tab', {
      name: 'Person',
    });
    await expect(firstTab).toBeInTheDocument();
    await expect(firstTab).toHaveAttribute('aria-selected', 'true');
    await expect(firstTab).toHaveAttribute('tabIndex', '0');
    const secondTab = canvas.getByRole('tab', {
      name: 'Bedrift',
    });
    await userEvent.click(secondTab);
    await expect(firstTab).toHaveAttribute('aria-selected', 'false');
    await expect(firstTab).toHaveAttribute('tabIndex', '-1');
  },
} satisfies Story;
