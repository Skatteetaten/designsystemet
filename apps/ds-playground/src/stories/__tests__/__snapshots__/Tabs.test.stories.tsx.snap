// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Tabs/Tabs Defaults play-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="_tabList_ibyzk_1  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r0:-tab1"
              class="_tab_1dx22_1   _tab_active_1dx22_27  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r0:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r0:-tab2"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r0:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r0:-tab3"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r0:-tab3"
      >
        Organisasjon
      </button>
    </div>
    <div id="ds-tab-panel-:r0:-tab1"
         class="_panel_1bj6c_1 _panel_active_1bj6c_4 "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r0:-tab1"
    >
      Tabs.Panel Person
    </div>
    <div id="ds-tab-panel-:r0:-tab2"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r0:-tab2"
         hidden
    >
      Tabs.Panel Bedrift
    </div>
    <div id="ds-tab-panel-:r0:-tab3"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r0:-tab3"
         hidden
    >
      Tabs.Panel Organisasjon
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithAriaRolesTabindex play-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="_tabList_ibyzk_1  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r6:-tab1"
              class="_tab_1dx22_1   _tab_active_1dx22_27  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r6:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r6:-tab2"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r6:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r6:-tab3"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r6:-tab3"
      >
        Organisasjon
      </button>
    </div>
    <div id="ds-tab-panel-:r6:-tab1"
         class="_panel_1bj6c_1 _panel_active_1bj6c_4 "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r6:-tab1"
    >
      Tabs.Panel Person
    </div>
    <div id="ds-tab-panel-:r6:-tab2"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r6:-tab2"
         hidden
    >
      Tabs.Panel Bedrift
    </div>
    <div id="ds-tab-panel-:r6:-tab3"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r6:-tab3"
         hidden
    >
      Tabs.Panel Organisasjon
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div id="htmlId"
       class="dummyClassname"
       lang="nb"
       data-testid="123ID"
  >
    <div class="_tabList_ibyzk_1  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r2:-tab1"
              class="_tab_1dx22_1   _tab_active_1dx22_27  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r2:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r2:-tab2"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r2:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r2:-tab3"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r2:-tab3"
      >
        Organisasjon
      </button>
    </div>
    <div id="ds-tab-panel-:r2:-tab1"
         class="_panel_1bj6c_1 _panel_active_1bj6c_4 "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r2:-tab1"
    >
      Tabs.Panel Person
    </div>
    <div id="ds-tab-panel-:r2:-tab2"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r2:-tab2"
         hidden
    >
      Tabs.Panel Bedrift
    </div>
    <div id="ds-tab-panel-:r2:-tab3"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r2:-tab3"
         hidden
    >
      Tabs.Panel Organisasjon
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithDefaultValue play-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="_tabList_ibyzk_1  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r4:-tab1"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r4:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r4:-tab2"
              class="_tab_1dx22_1   _tab_active_1dx22_27  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r4:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r4:-tab3"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r4:-tab3"
      >
        Organisasjon
      </button>
    </div>
    <div id="ds-tab-panel-:r4:-tab1"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r4:-tab1"
         hidden
    >
      Tabs.Panel Person
    </div>
    <div id="ds-tab-panel-:r4:-tab2"
         class="_panel_1bj6c_1 _panel_active_1bj6c_4 "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r4:-tab2"
    >
      Tabs.Panel Bedrift
    </div>
    <div id="ds-tab-panel-:r4:-tab3"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r4:-tab3"
         hidden
    >
      Tabs.Panel Organisasjon
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithMultiline smoke-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="_tabList_ibyzk_1 _tabList_multiline_ibyzk_6 "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r7:-tab1"
              class="_tab_1dx22_1   _tab_active_1dx22_27  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r7:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r7:-tab2"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r7:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r7:-tab3"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r7:-tab3"
      >
        Ubehandlede saker (+99)
      </button>
      <button id="ds-tab-id-:r7:-tab4"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r7:-tab4"
      >
        Behandlede saker (1)
      </button>
    </div>
    <div id="ds-tab-panel-:r7:-tab1"
         class="_panel_1bj6c_1 _panel_active_1bj6c_4 "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r7:-tab1"
    >
      Tabs.Panel - Person
    </div>
    <div id="ds-tab-panel-:r7:-tab2"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r7:-tab2"
         hidden
    >
      Tabs.Panel - Bedrift
    </div>
    <div id="ds-tab-panel-:r7:-tab3"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r7:-tab3"
         hidden
    >
      Tabs.Panel - Ubehandlede saker (+99)
    </div>
    <div id="ds-tab-panel-:r7:-tab4"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r7:-tab4"
         hidden
    >
      Tabs.Panel - Behandlede
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithRef play-test 1`] = `
<div data-test-block="true">
  <div class
       id="dummyIdForwardedFromRef"
  >
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithTabClick play-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="_tabList_ibyzk_1  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r8:-tab1"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r8:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r8:-tab2"
              class="_tab_1dx22_1   _tab_active_1dx22_27  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r8:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r8:-tab3"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r8:-tab3"
      >
        Organisasjon
      </button>
    </div>
    <div id="ds-tab-panel-:r8:-tab1"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r8:-tab1"
         hidden
    >
      Tabs.Panel Person
    </div>
    <div id="ds-tab-panel-:r8:-tab2"
         class="_panel_1bj6c_1 _panel_active_1bj6c_4 "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r8:-tab2"
    >
      Tabs.Panel Bedrift
    </div>
    <div id="ds-tab-panel-:r8:-tab3"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r8:-tab3"
         hidden
    >
      Tabs.Panel Organisasjon
    </div>
  </div>
</div>
`;

exports[`Tester/Tabs/Tabs WithValue play-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="_tabList_ibyzk_1  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r5:-tab1"
              class="_tab_1dx22_1     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r5:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r5:-tab2"
              class="_tab_1dx22_1   _tab_active_1dx22_27  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r5:-tab2"
      >
        Bedrift
      </button>
    </div>
    <div id="ds-tab-panel-:r5:-tab1"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r5:-tab1"
         hidden
    >
      Tabs.Panel - Person
    </div>
    <div id="ds-tab-panel-:r5:-tab2"
         class="_panel_1bj6c_1 _panel_active_1bj6c_4 "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r5:-tab2"
    >
      Tabs.Panel - Bedrift
    </div>
  </div>
  <button class="_button_6ybyo_1 _button_primary_6ybyo_1   "
          type="button"
  >
    <span class="_buttonText_6ybyo_69">
      ToggleTab
    </span>
  </button>
</div>
`;

exports[`Tester/Tabs/Tabs WithVariantCompact smoke-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="_tabList_ibyzk_1  "
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r3:-tab1"
              class="_tab_1dx22_1 _tab_compact_1dx22_43  _tab_active_1dx22_27  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r3:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r3:-tab2"
              class="_tab_1dx22_1 _tab_compact_1dx22_43    "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r3:-tab2"
      >
        Bedrift
      </button>
      <button id="ds-tab-id-:r3:-tab3"
              class="_tab_1dx22_1 _tab_compact_1dx22_43    "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r3:-tab3"
      >
        Organisasjon
      </button>
    </div>
    <div id="ds-tab-panel-:r3:-tab1"
         class="_panel_1bj6c_1 _panel_active_1bj6c_4 "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r3:-tab1"
    >
      Tabs.Panel Person
    </div>
    <div id="ds-tab-panel-:r3:-tab2"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r3:-tab2"
         hidden
    >
      Tabs.Panel Bedrift
    </div>
    <div id="ds-tab-panel-:r3:-tab3"
         class="_panel_1bj6c_1  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r3:-tab3"
         hidden
    >
      Tabs.Panel Organisasjon
    </div>
  </div>
</div>
`;
